e2e/node/resource_node.go

```diff
@@ -410,11 +437,14 @@ func resourceUpdateNode(ctx context.Context, d *schema.ResourceData, m interface
 	nodeId := d.Id()
 	project_id := d.Get("project_id").(string)
 	location := d.Get("location").(string)
+	status := d.Get("status").(string)
+	if status == "Failed" {
```

Comment:
the value "Failed" must be defined in a constant. If we have pre-defined states, then those should be defined in a constant. All such states must be kept in a single array or hashmap and fetched from there.